test/test_pso.py

```diff
@@ -132,6 +132,26 @@ def test_stopping_no_criterion(self):
         opt.set_max_unchanged_iterations(None)
         self.assertRaises(ValueError, opt.run)
 
+    def test_logpdf(self):
```

Comment:
Just curious, are we going to test this for other optimisation algorithms?

Reply:
Yeah it should become part of the the other methods too, but as there are other tickets open about this I didn't want to do it here!